internal/releasesources/release_sources.go

```diff
@@ -0,0 +1,413 @@
+				if err != nil {
+					return err
+				}
+
+				if switchedArch {
+					if arch, ok := cpuArchMapFromAPIArch[cpuArchitecture]; ok {
+						cpuArchitecture = arch
+					}
+				}
+
+				for _, node := range graph.Nodes {
+					if strings.HasPrefix(node.Version, openshiftVersion) {
+						// Got a match for OpenShiftReleaseImage
+						err = createOrUpdateRelease(db, channel, node.Version, cpuArchitecture)
+						if err != nil {
+							return err
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return nil
+}
+
+func deletePreReleaseVersions(db *gorm.DB, openshiftMajorMinorVersion string) error {
+	var (
+		releasesToDelete []common.OpenshiftVersion
+		releases         []common.OpenshiftVersion
+	)
+	err := db.Model(&common.OpenshiftVersion{}).
+		Find(&releases).Error
+	if err != nil {
+		return err
+	}
+
+	for _, release := range releases {
+		majorMinorVersion, err := common.GetMajorMinorVersion(*release.OpenshiftVersion.DisplayName)
+		if err != nil {
+			return err
+		}
+
+		if *majorMinorVersion == openshiftMajorMinorVersion && release.Channel == "candidate" {
+			releasesToDelete = append(releasesToDelete, release)
+		}
+	}
+	// Nothing to delete
+	if len(releasesToDelete) == 0 {
+		return nil
+	}
+
+	return db.Where("TRUE").Delete(&releasesToDelete).Error
+}
+
+func setSupportLevel(db *gorm.DB, version Version) error {
+	var (
+		releases []common.OpenshiftVersion
+	)
+	err := db.Model(&common.OpenshiftVersion{}).
+		Find(&releases).Error
+	if err != nil {
+		return err
+	}
+
+	for _, release := range releases {
+		releaseCopy := release
+		majorMinorVersion, err := common.GetMajorMinorVersion(*release.OpenshiftVersion.DisplayName)
+		if err != nil {
+			return err
+		}
+
+		if *majorMinorVersion == version.Name {
+			releaseCopy.SupportLevel = &version.Type
+			err = db.Save(&releaseCopy).Error
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+func setDefaultRelease(db *gorm.DB) error {
+	var (
+		defaultCandidatesStableReleases []common.OpenshiftVersion
+		err                             error
+		lessThan                        bool
+	)
+	err = db.Model(&common.OpenshiftVersion{}).
+		Where("channel = 'stable' AND ? = ANY(cpu_architectures)", defaultCPUArchitecture).
+		Find(&defaultCandidatesStableReleases).Error
+	if err != nil {
+		return err
+	}
+
+	if len(defaultCandidatesStableReleases) == 0 {
+		return fmt.Errorf("no stable releases with cpu architecture %s found", defaultCPUArchitecture)
+	}
+
+	latestReleaseVersion := defaultCandidatesStableReleases[0].DisplayName
+	latestStableRelease := defaultCandidatesStableReleases[0]
+
+	for _, release := range defaultCandidatesStableReleases {
+		lessThan, err = common.BaseVersionLessThan(*release.DisplayName, *latestReleaseVersion)
+		if err != nil {
+			return err
+		}
+
+		if lessThan {
+			latestReleaseVersion = release.DisplayName
+			latestStableRelease = release
+		}
+	}
+
+	// Changing to new default
+	err = db.Model(&common.OpenshiftVersion{}).
```

Comment:
Not critical, but this should run in a transaction together with setting a new default, and possibly more.

Reply:
Do you mean I should select the table before everything and then use it everywhere? I would need to select it again every time I am ending a transaction with `DELETE/UPDATE/...` no?